docs/plugins/lifecycle.md

```diff
@@ -17,9 +20,10 @@ The following states are available:
 | State         | Description                                 |
 | ------------- |:-------------------------------------------:|
 | init          | Application is initialising                 |
+| ready         | Application is ready for user interaction   |
 | active        | Application is running, and in focus        |
 | background    | Application is running, not in focus        |
-| pause         | Application is paused, not in focus         |
+| pause         | Application is pausing, not in focus        |
```

Comment:
I think I'd prefer these to be called `pausing` and `closing` so that it's clear that these are transitional. I suppose the same goes for `initializing`.

Reply:
yes, I guess that's a valid point .. @wouterlucas what do you think?

Reply:
What do you mean by transitional, here, @cletusboone? Aren't all state machines are transitional?

Reply:
I guess they are, but there's a subtle difference in the wording for each state. Notice that `active` says that the "Application **is** running," but the `close` state says that it _is closing_.

The difference being that the app will receive a signal from the control plane to close itself, and the app might execute logic. It will not know that it is closed. It can only know that it _should_ close and what it can tell the control plane is that it _is closing._ So if you query the state at that point, we might as well be semantically correct about what's actually going on.

For `pause`, maybe it doesn't matter since the app can presumably know that it is paused or that it _will pause_ and it can report this state to the control plane.